editor/inserter/index.js

```diff
@@ -1,18 +1,31 @@
+/**
+ * External dependencies
+ */
+import { groupBy } from 'lodash';
+
 const Inserter = () => {
 	const blocks = wp.blocks.getBlocks();
+	const blocksByCategory = groupBy( blocks, 'category' );
+	const categories = wp.blocks.getCategories();
 
 	return (
 		<div className="inserter">
 			<div className="inserter__arrow" />
 			<div className="inserter__content">
-				<div className="inserter__Category-blocks">
-					{ blocks.map( ( { slug, title, icon } ) => (
-						<div key={ slug } className="inserter__block">
-							<span className={ 'dashicons dashicons-' + icon } />
-							{ title }
+				{ categories
```

Comment:
>Already answered this above :)

Whoops! I had sensed I'd already remarked but missed the collapsed comment when looking through the conversation history.

To your point: Fair to call it a difference of opinion, but aside from performance wins, I'd also disagree that `reduce` is any less legible. To me it's simply looping to aggregate a value with conditions on (or formatting applied to) the values included.

```js
categories.reduce( ( memo, category ) => {
	if ( !! blocksByCategory[ category.slug ] ) {
		memo.push(
			<div key={ category.slug } className="inserter__Category-blocks">
				<div className="inserter__separator">{ category.title }</div>
				{ blocksByCategory[ category.slug ].map( ( { slug, title, icon } ) => (
					<div key={ slug } className="inserter__block">
						<span className={ 'dashicons dashicons-' + icon } />
						{ title }
					</div>
				) ) }
			</div>
		);
	}

	return memo;
}, [] )
```